test/unit/communication/test_comm_sender.cpp

```diff
@@ -41,10 +41,7 @@ void test_transform_mpi() {
 
   auto send = just(send_buf.data(), size, dtype, send_rank, tag, comm) | transformMPI(MPI_Isend);
   auto recv = just(recv_buf.data(), size, dtype, recv_rank, tag, comm) | transformMPI(MPI_Irecv);
-  sync_wait(when_all(std::move(send), std::move(recv)) | then([](int e1, int e2) {
-              DLAF_MPI_CHECK_ERROR(e1);
-              DLAF_MPI_CHECK_ERROR(e2);
-            }));
+  sync_wait(when_all(std::move(send), std::move(recv)) | then([]() {}));
```

Comment:
If we're really planning on leaving out the return value of the function passed to `transform_mpi`, you can remove the `then` from here (same further down). It serves no purpose.

Reply:
```suggestion
  sync_wait(when_all(std::move(send), std::move(recv)));
```